components/builder-jobsrv/src/server/worker_manager.rs

```diff
@@ -451,6 +462,31 @@ impl WorkerMgr {
         Ok(())
     }
 
+    fn is_job_complete(&mut self, job_id: u64) -> Result<bool> {
+        let mut req = jobsrv::JobGet::new();
+        req.set_id(job_id);
+
+        let ret = match self.datastore.get_job(&req)? {
+            Some(job) => {
+                match job.get_state() {
+                    jobsrv::JobState::Pending |
+                    jobsrv::JobState::Processing |
+                    jobsrv::JobState::Dispatched => false,
+                    _ => true,
```

Comment:
I would recommend explicitly exhausting all matches so the compiler warns somebody about this function if they add an additional state which may be considered an "incomplete" state